server/server.ts

```diff
@@ -119,7 +116,29 @@ app.delete('/api/catalog/cart/:productId', async (req, res, next) => {
     const deletedCartItem = results.rows[0];
     if (!deletedCartItem)
       throw new ClientError(404, `error could not find ${productId}`);
-    res.status(204);
+    res.sendStatus(204);
+  } catch (error) {
+    next(error);
+  }
+});
+
+app.put('/api/catalog/cart/:productId', async (req, res, next) => {
+  try {
+    const { productId } = req.params;
+    const { quantity } = req.body;
+    const sql = `
+    update "cartItems"
+    set "quantity" = $1
+    where "productId" =$2
+    returning*
+    `;
+    const params = [quantity, productId];
+    const results = await db.query(sql, params);
+    const updatedProduct = results.rows[0];
+    if (!updatedProduct) {
+      throw new ClientError(404, `error ${productId} does not exist`);
+    }
+    res.sendStatus(200).json(updatedProduct);
```

Comment:
This is wrong. (I think we fixed it later.) `.sendStatus` actually sends a response, so `.json` will then fail. You use `.status()` to just set the status and send the response in a later call. Also, since 200 is default, you could just use `res.json()`.